### PR TITLE
Salt cogs

### PR DESCRIPTION
**Pull request by ScalarVector1**
This PR implements the salt cogs item, which increases your sentry's speed temporarily when summoned, and increases your sentry placement speed.

### CODE REVIEW

File: Content/Items/Beach/Acessories.SaltCogs.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		private void SpeedUp(Projectile projectile)
+		{
+			Player owner = Main.player[projectile.owner];
+
+			if (owner != null && Equipped(owner))
+			{
+				if (projectile.timeLeft == (Projectile.SentryLifeTime - 1))
+					projectile.extraUpdates += 1;
+
+				if (projectile.timeLeft == (Projectile.SentryLifeTime - 240))
```
Comment (TripleFate) [inline Content/Items/Beach/Acessories.SaltCogs.cs:36]: isn't 240 4 seconds while the description says 2 seconds? I'd say move this 240 into a const and have the number in the description be programmatically assigned

File: Content/Items/Beach/Acessories.SaltCogs.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			}
+		}
+
+		private void DrawCog(Projectile projectile, Color lightColor)
+		{
+			Player owner = Main.player[projectile.owner];
+
+			if (owner != null && Equipped(owner))
+			{
+				if (projectile.timeLeft >= (Projectile.SentryLifeTime - 240))
```
Comment (TripleFate) [inline Content/Items/Beach/Acessories.SaltCogs.cs:47]: especially when the 240 gets reused like this makes it easier to balance adjust in the future and gives more meaning to this hanging magic number

**Review by TripleFate — APPROVED** [review]
Nothing wrong with this. As a future guideline though I'd recommend we move magic numbers like the "240" in this one into consts so its easier for non-programmers to understand some features and make it easier to read code and make balance adjustments on some numbers

File: Content/Items/Beach/Acessories.SaltCogs.cs (hunk 1 above)
Comment (TripleFate) [reply Content/Items/Beach/Acessories.SaltCogs.cs:36]: oh wait does extraUpdates make SentryLifeTime count more times too? that would explain this

Comment (ScalarVector1) [reply Content/Items/Beach/Acessories.SaltCogs.cs:36]: yep, 120 would be 1 second as extraUpdates causes it to tick down twice as fast.

**Pull request merged**